text/0000-target-tier-policy.md

```diff
@@ -0,0 +1,374 @@
+- All tier 2 requirements apply.
+
+A tier 1 target may be demoted if it no longer meets these requirements. Any
+proposal for demotion or removal will be communicated widely to the Rust
+community, both when initially proposed and before being dropped from a stable
+release. A tier 1 target is highly unlikely to be directly removed without
+first being demoted to tier 2 or tier 3.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+The set of approving teams for each tier arose out of discussion with the
+various teams involved with aspects of the Rust project impacted by new
+targets.
+
+Policies that require the approval of multiple teams could instead require a
+core team approval. This would have the advantage of reducing the number of
+people involved in the final approval, but would put more coordination effort
+on the core team and the various team leads to ensure that the individual teams
+approve. As another alternative, we could separate the individual team
+approvals (into separate issues or separate rfcbot polls), to simplify checking
+for consensus and reduce diffusion of responsibility; however, this could also
+increase the resulting complexity and result in discussions in multiple places.
+
+We could introduce specific time requirements for the amount of time a target
+must spend at a tier before becoming eligible for promotion to a higher tier.
+
+# Prior art
+[prior-art]: #prior-art
+
+This attempts to formalize and document Rust policy around targets and
+architectures. Some requirements of such a policy appear on the [Rust Platform
+Support page](https://forge.rust-lang.org/release/platform-support.html).
+
+Future expansions of such policy may find requirements from other communities
+useful as examples, such as Debian's [arch
+policy](https://release.debian.org/bullseye/arch_policy.html) and [archive
+criteria](https://ftp-master.debian.org/archive-criteria.html).
+
+Other precedents for tiered target support include [Firefox's supported build
+targets](https://firefox-source-docs.mozilla.org/build/buildsystem/supported-configurations.html),
+[node.js supported
+platforms](https://github.com/nodejs/node/blob/master/BUILDING.md#supported-platforms),
+and [GHC's platform
+support](https://gitlab.haskell.org/ghc/ghc/-/wikis/platforms).
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+- How should we track the maintainers of a target, so that we can page them if
+  we need an issue addressed involving that target?
+  - We could use github teams, which could be directly mentioned in an issue or
+    PR to get their attention. However, this would also put a "Member" badge on
+    the members of those teams, which may give an unwarranted appearance of
+    official status.
+  - We could track them in a document posted somewhere, and manually copy-paste
+    the list to ping them.
+  - We could add them as a "marker team" (e.g. `target-xyz`) in the
```

Comment:
We've kind of started already doing this with the notification groups for targets (Windows, ARM) in the compiler.